src/main/java/edu/iis/powp/command/factory/CommandFactoryWindow.java

```diff
@@ -0,0 +1,268 @@
+        leftPanel.setLayout(gridBagLayout);
+        GridBagConstraints leftPanelConstraints = new GridBagConstraints();
+        leftPanelConstraints.fill = 1;
+        leftPanelConstraints.gridheight = 1;
+        leftPanelConstraints.gridx = 0;
+        leftPanelConstraints.gridy = 0;
+        leftPanelConstraints.weighty = 1;
+        this.getContentPane().add(leftPanel, leftPanelConstraints);
+
+        JList jList = new JList();
+        JScrollPane listScrollPane = new JScrollPane();
+        listScrollPane.setVerticalScrollBarPolicy(ScrollPaneConstants.VERTICAL_SCROLLBAR_ALWAYS);
+        listScrollPane.setViewportView(jList);
+
+        listScrollPane.setPreferredSize(new Dimension(0, 520/2));
+        DefaultListModel<IPlotterCommand> jListModel = new DefaultListModel<>();
+        jList.setModel(jListModel);
+        GridBagConstraints jListConstraints = new GridBagConstraints();
+        jListConstraints.fill = 1;
+        jListConstraints.gridheight = 1;
+        jListConstraints.gridx = 0;
+        jListConstraints.gridy = 1;
+        jListConstraints.weighty = 1;
+        this.getContentPane().add(listScrollPane, jListConstraints);
+
+
+        JComboBox<ConstructorDecorator> basicComboBox = new JComboBox<>();
+        basicComboBox.setPreferredSize(new Dimension(COMMON_WIDTH2, COMMON_HEIGHT));
+        for (Constructor<? extends IPlotterCommand> constructor : commandRegistry.getRegisteredBasicCommands()) {
+            basicComboBox.addItem(new ConstructorDecorator(constructor));
+        }
+        GridBagConstraints basicComboBoxConstraints = new GridBagConstraints();
+        basicComboBoxConstraints.gridx = 0;
+        basicComboBoxConstraints.gridy = 0;
+        leftPanel.add(basicComboBox,basicComboBoxConstraints);
+
+        JButton basicAddButton = new JButton("Add");
+        basicAddButton.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(final ActionEvent e) {
+                JFrame frame = new JFrame();
+                frame.setBounds(100,100,200,100);
+                ConstructorDecorator selectedItem = (ConstructorDecorator) basicComboBox.getSelectedItem();
+                Constructor<? extends IPlotterCommand> constructor = selectedItem.getConstructor();
+                frame.setLayout(new GridLayout(constructor.getParameterCount() + 1, 2));
+                Map<Parameter, JTextField> textFields = new LinkedHashMap<>();
+                for (Parameter parameter : constructor.getParameters()) {
+                    JLabel label = new JLabel(parameter.getName());
+                    frame.add(label);
+                    JTextField field = new JTextField();
+                    frame.add(field);
+                    textFields.put(parameter, field);
+                }
+                JButton button = new JButton("Add");
+                button.addActionListener(new ActionListener() {
+                    @Override
```

Comment:
FIX TO MERGE: Again: 2nd level of embedding logic in GUI!
None GUI logic can be extracted as a model (implementing a specific interface). The model can influence view synchronously (operating on GUI using some interface) or asynchronously (using observer pattern).